cmake/SlicerRadiomicsAddCLI.cmake

```diff
@@ -0,0 +1,101 @@
+################################################################################
+#
+#  Program: SlicerRadiomics
+#
+#  Copyright (c) Kitware Inc.
+#
+#  See LICENSE.txt
+#
+#  This file was originally developed by Jean-Christophe Fillion-Robin, Kitware Inc.
+#  and was partially funded by NIH grant 1U24CA194354-01
+#
+################################################################################
+
+# This CMake module defines the function SlicerRadiomicsAddCLI.
+
+#
+# SlicerRadiomicsAddCLI
+# =====================
+#
+# Add a shell script-based Slicer CLI to the build system.
+#
+# SlicerRadiomicsAddCLI(
+#   NAME <module_name>
+#   )
+#
+# NAME This is the name of the CLI to configure and install. It expects the following
+#      files to exist in the current directory:
+#        <module_name>
+#        <module_name>.bat
+#        <module_name>.xml
+#
+# Notes:
+#
+#  * The function adds a custom target named ``Copy<module_name>Scripts``
+#    responsible to copy module files into ``SlicerExecutionModel_DEFAULT_CLI_RUNTIME_OUTPUT_DIRECTORY``.
+#
+#  * The function adds an install rule associated with ``RuntimeLibraries`` component that will install
+#    the module files into ``SlicerExecutionModel_DEFAULT_CLI_INSTALL_RUNTIME_DESTINATION``.
+#
+#  * The list of module files is composed differently based on the platform:
+#    - Unix: <module_name> and <module_name>.xml
+#    - Windows: <module_name>.bat and <module_name>.xml
+#
+function(SlicerRadiomicsAddCLI)
+  set(options
+    )
+  set(oneValueArgs
+    NAME
+    )
+  set(multiValueArgs
+    )
+  cmake_parse_arguments(MY
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+    )
+  message(STATUS "Configuring SEM CLI module: ${MY_NAME}")
+
+  # Sanity checks
+  set(expected_defined_vars
+    SlicerExecutionModel_DEFAULT_CLI_RUNTIME_OUTPUT_DIRECTORY
+    SlicerExecutionModel_DEFAULT_CLI_INSTALL_RUNTIME_DESTINATION
+    )
+  foreach(var IN LISTS expected_defined_vars)
+    if(NOT DEFINED ${var})
+      message(FATAL_ERROR "Variable ${var} is not defined !")
+    endif()
+  endforeach()
+
+  set(expected_parameters
+    NAME
+    )
+  foreach(param IN ITEMS NAME)
+    if(NOT MY_${param})
+      message(FATAL_ERROR "Parameter ${param} is expected !")
+    endif()
+  endforeach()
+
+  set(cli_script "${MY_NAME}")
+  if(WIN32)
+    set(cli_script "${MY_NAME}.bat")
+  endif()
```

Comment:
@jcfr, If I'm reading this correctly, it will install the `SlicerRadiomicsCLI.bat` if it is built on windows and `SlicerRadiomicsCLI` on other platforms. However, the `.bat` script's function is to start the `SlicerRadiomicsCLI` script, so that on will also have to be installed on Windows.